[1488] Support frequency nominal dim in consolidate functions

dbashford-NOAA opened the pull request:
Resolves #1488 

- Use a generalized `dim_0` so frequency nominal can be supported in place of channel
- Added testing for the swapped dims
- Update tests to use fixture for the test data folder

dbashford-NOAA committed:
[1488] Support frequency nominal dim in consolidate functions and update tests

# Conflicts:
#	echopype/tests/consolidate/test_add_depth.py

dbashford-NOAA committed:
[1488] Support frequency nominal dim in consolidate functions and update tests

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

dbashford-NOAA committed:
[1486] Support computing MVBS with frequency_nominal and fix some tests [all tests ci] (#1517)

* [1486] Support computing MVBS with frequency_nominal and fix some tests

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* [1486] Use Dataset.sizes instaed of Dataset.dims

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump actions/download-artifact from 4 to 5 (#1529)

Bumps [actions/download-artifact](https://github.com/actions/download-artifact) from 4 to 5.
- [Release notes](https://github.com/actions/download-artifact/releases)
- [Commits](https://github.com/actions/download-artifact/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/download-artifact
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump actions/checkout from 4 to 5 (#1528)

Bumps [actions/checkout](https://github.com/actions/checkout) from 4 to 5.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump actions/cache from 4.2.3 to 4.2.4 (#1527)

Bumps [actions/cache](https://github.com/actions/cache) from 4.2.3 to 4.2.4.
- [Release notes](https://github.com/actions/cache/releases)
- [Changelog](https://github.com/actions/cache/blob/main/RELEASES.md)
- [Commits](https://github.com/actions/cache/compare/v4.2.3...v4.2.4)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-version: 4.2.4
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate (#1526)

updates:
- [github.com/pre-commit/pre-commit-hooks: v5.0.0 → v6.0.0](https://github.com/pre-commit/pre-commit-hooks/compare/v5.0.0...v6.0.0)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

LOCEANlloydizard committed:
Update pr.yaml to run full test suite for all PR (#1524)

This PR updates the pr.yaml GitHub Actions workflow to always run the full test suite for all pull requests, regardless of the PR title, by removing the conditional logic that previously limited full test runs to PRs titled with [all tests ci].

leewujung commented on echopype/consolidate/api.py:
```diff
@@ -455,12 +456,14 @@ def add_splitbeam_angle(
 
-    # check that source_Sv at least has a channel dimension
-    if "channel" not in source_Sv.variables:
-        raise ValueError("The input source_Sv Dataset must have a channel dimension!")
+    dim_0 = list(source_Sv.sizes.keys())[0]
 
-    # Select ds_beam channels from source_Sv
-    ds_beam = echodata[ed_beam_group].sel(channel=source_Sv["channel"].values)
+    if dim_0 in ["channel", "frequency_nominal"]:
+        ds_beam = echodata[ed_beam_group].sel({dim_0: source_Sv[dim_0].values})
+    else:
+        raise ValueError(
+            "The input source_Sv Dataset must have a channel or frequency_nominal dimension!"
+        )
```
In this implementation `dim_0` can be anything that is in the first position (index 0) in the `ds` dataset keys. In theory`ds` can come from different sources, so there is no guarantee that the first key would be `channel` or `frequency_nominal`. Since you check `if dim_0 in ["channel", "frequency_nominal"]` downstream anyways, how about checking if the dimension contains one of these two options before proceeding?

Also, the current implementation
```
            if echodata["Sonar/Beam_group1"][dim_0].equals(ds[dim_0]):
                beam_group_name = "Beam_group1"
            else:
                beam_group_name = "Beam_group2"
```
is a weird setup in terms of Beam_groupX selection. This is a larger issue so no immediate actions required under this PR, but I will note this in a separate issue.

leewujung commented on echopype/tests/consolidate/test_consolidate_integration.py:
```diff
@@ -286,6 +286,147 @@ def test_add_splitbeam_angle(sonar_model, test_path_key, raw_file_name, test_pat
+        "ek80_CW_power",
+    ],
+)
+def test_add_splitbeam_angle_w_dim_swap(sonar_model, test_path_key, raw_file_name, test_path,
+                             paths_to_echoview_mat, waveform_mode, encode_mode,
+                             pulse_compression, to_disk):
```
This new test is largely identical with the existing `test_add_splitbeam_angle`, and the content of that test was to make sure the split beam angle computation is correct. I think it's a bit meddling the water to repeat the same thing here, since dimension swap is not part of the split beam calculate.

Could you slim this down to just testing the dimension swap?

dbashford-NOAA replied:
There is already a test for the `swap_dims_channel_frequency` and I didn't make any changes to that function but do you think it needs more coverage? This test was meant to cover the changes I made to 'test_add_splitbeam_angle' and make sure the split beam angle computation was correct when the dims are swapped. Maybe it would be better to just add a dim swap test case to the existing `test_add_splitbeam_angle` test?

leewujung replied:
adding one test to go through the `swap_dims_channel_frequency` step and then `add_splitbeam_angle` would be great. I think making it a separate test from `test_add_splitbeam_angle` which tests the function outputs would be a good idea.

leewujung reviewed (changes requested):
@dbashford-NOAA : Thanks for the PR. Sorry it took a while for me to get to this.

I made a few suggestions that I think will improve the robustness and the tests.

Could you also add a test for `add_location` to make sure things flow through even with dimension swap? Thanks.

Please request my review once you're done with the above.

dbashford-NOAA committed:
[1488] Create consolidate tests that include dim swapping

dbashford-NOAA committed:
[1488] Create consolidate tests that include dim swapping

dependabot[bot] committed:
chore(deps): bump pypa/gh-action-pypi-publish in /.github/workflows (#1534)

Bumps [pypa/gh-action-pypi-publish](https://github.com/pypa/gh-action-pypi-publish) from 1.12.4 to 1.13.0.
- [Release notes](https://github.com/pypa/gh-action-pypi-publish/releases)
- [Commits](https://github.com/pypa/gh-action-pypi-publish/compare/v1.12.4...v1.13.0)

---
updated-dependencies:
- dependency-name: pypa/gh-action-pypi-publish
  dependency-version: 1.13.0
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump actions/setup-python from 5.5.0 to 6.0.0 (#1535)

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 5.5.0 to 6.0.0.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v5.5.0...v6.0.0)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-version: 6.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

leewujung commented on echopype/tests/consolidate/test_add_location.py:
```diff
@@ -184,6 +184,75 @@ def _tests(ds_test, location_type, nmea_sentence=None):
+
+    ds = ep.calibrate.compute_Sv(echodata=ed, env_params=env_params)
+
+    ds = ep.consolidate.swap_dims_channel_frequency(ds)
+    for group in ed["Sonar"]['beam_group'].values:
+        ed[f"Sonar/{group}"] = ep.consolidate.swap_dims_channel_frequency(ed[f"Sonar/{group}"])
+
+    ds_all = ep.consolidate.add_location(ds=ds, echodata=ed)
```
(Somehow I missed your response...)

Yes I think all `ep.consolidate` functions should work that way. Basically the functions should work with either `channel` or `frequency_nominal` as a coordinate.

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate (#1538)

updates:
- [github.com/psf/black: 25.1.0 → 25.9.0](https://github.com/psf/black/compare/25.1.0...25.9.0)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

LOCEANlloydizard committed:
Add weill to codespell ignore list (#1543)

Extend .pre-commit-config.yaml to ignore the word "weill", which is used as a method in the shoal detection module. If not, codespell incorrectly flags and auto-corrects it to "will".

LOCEANlloydizard committed:
Implementations of basic seafloor detection in mask subpackage [all tests ci] (#1521)

* Implementations of basic seafloor detection in mask subpackage; example notebook coming soon.

I have tested two possible ways to integrate the seafloor detection into the current workflow. Both functions are currently implemented in the mask subpackage (the second one is temporarily located in the seafloor_detection subpackage within mask).

An example notebook demonstrating and testing the feature will be submitted soon to the echopype-examples repository for review.

* Add detect_bottom() dispatcher function for bottom line detection Description:

* Introduced detect_bottom() in mask/api.py to act as a unified interface for bottom detection methods
* Replaced the earlier class-based approach with simpler functional structure under mask/seafloor_detection
* Output is now a 1D DataArray (no channel dim), with one depth per ping

* Added Blackwell detection method (adapted from echopy)

Adds a new bottom detection method (detect_blackwell), based on the approach by Blackwell et al. (2019) and adapted from the Echopy package. It uses both Sv and split-beam angles (angle_alongship, angle_athwartship) to detect the seafloor and returns one bottom depth per ping.

Other changes:

    * Renamed the surface_skip argument to bin_skip_from_surface in detect_basic() to make it clearer.

* Added test for seafloor basic and blackwell methods

Added four basic tests for basic and blackwell methods

* Update test_mask.py

updated the test "test_blackwell_vs_basic_close" to use local file

* Addresses review feedback

- Add detailed docstrings to:
  - detect_seafloor (now documents per-method args)
  - bottom_blackwell
  - bottom_basic
- Rename registry from METHODS -> METHODS_BOTTOM to scope it to bottom detection.
- Replace local lin/log helpers with echopype.utils.compute._log2lin / _lin2log; remove duplicates.
- Rename functions to make purpose explicit:
  - detect_basic      -> bottom_basic
  - detect_blackwell  -> bottom_blackwell
- Update associated files/exports to match new names.

Refs: PR #1521 (addresses review feedback)

* Apply suggestions from code review

docstring modifications

Co-authored-by: Wu-Jung Lee <leewujung@gmail.com>

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: Wu-Jung Lee <leewujung@gmail.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

dbashford-NOAA committed:
[1488] Swap beam group dims in consolidate functions if needed

dbashford-NOAA committed:
[1488] Swap beam group dims in consolidate functions if needed

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

dbashford-NOAA committed:
[1488] Fix f-string placeholder error

dbashford-NOAA committed:
[1488] Fix f-string placeholder error

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

dbashford-NOAA committed:
[1541] Remove python 3.10 support (#1546)

dependabot[bot] committed:
chore(deps): bump actions/cache from 4.2.4 to 4.3.0 (#1547)

Bumps [actions/cache](https://github.com/actions/cache) from 4.2.4 to 4.3.0.
- [Release notes](https://github.com/actions/cache/releases)
- [Changelog](https://github.com/actions/cache/blob/main/RELEASES.md)
- [Commits](https://github.com/actions/cache/compare/v4.2.4...v4.3.0)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-version: 4.3.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

LOCEANlloydizard committed:
Add Fielding and Matecho-like transient noise removal methods (#1544)

* Add Fielding and Matecho-like transient noise removal methods

This commit introduces two new transient noise removal methods:
* Fielding-based filter
* Matecho-style deep transient spike filter
* Add tests for both

For now, both methods are implemented in a new subpackage transient_noise.
The existing Ryan method remains in api.py. In a follow-up commit, we can either (a) move the Ryan method into the same subpackage, or (b) re-export these new methods from api.py for consistency.

The tests are also stored in their own test_transient_noise.py for now. They can easily be move to the test_noise.py, or vice-versa.

Closes #1352

* Updates to match review

changed function names for consistency and fixed bottom_var if its provided

* Add detailed docstring and rename dispatcher to detect_transient

- Added a detailed docstring for the transient-noise dispatcher
- Renamed dispatcher to detect_transient to match detect_seafloor and detect_shoal

* Updated the tests

Forgot to update the tests with new dispatcher name.

changed "mask_transient_noise_dispatch" to "detect_transient"

LOCEANlloydizard committed:
Shoals detection implementation [all tests ci] (#1525)

* Shoals detection implementation

Initial implementation of shoal detection in echopype, adapted from the echopy package

* Adding Weill method and tests for Weill and echoview methods

- Add Weill shoal detector to detect_shoals
- Add basic tests for Weill and Echoview methods
- No API breaks; mask returned as (ping_time, range_sample).

Refs #1523.

* Apply suggestions from code review

Co-authored-by: Wu-Jung Lee <leewujung@gmail.com>

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* resolve merge conflicts

Fix conflicts from upstream seafloor merge

* use 'weill' spelling in shoal tests

few typos not corrected

---------

Co-authored-by: Wu-Jung Lee <leewujung@gmail.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

LOCEANlloydizard committed:
Modifications to match xarray update to v2025.9.1 (#1550)

* Fix: update_platform attr assignment; pass engine to to_netcdf for grouped writes

* Update pr.yaml

* ci: make workflow_dispatch safe; skip changed-files on manual runs

* ci: drop pr.yaml changes for upstream PR

* Update after review: use update() and remove hardcoded NetCDF engine

- Switched back to using Dataset.update() for replacing variables (added back the comment clarifying that drop is not needed).
- Removed hardcoded "netcdf4" in utils/io.py and replaced with engine=engine, to align with SUPPORTED_ENGINES validation.

dbashford-NOAA committed:
[1488] Re add ed ds compare for add_depth

dbashford-NOAA committed:
[1488] Re add ed ds compare for add_depth

LOCEANlloydizard committed:
Update docstrings for transient noise and shoal detection functions

Update docstrings for transient noise and shoal detection functions, and add credit to Echopy for the original implementations.

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate (#1555)

updates:
- [github.com/PyCQA/isort: 6.0.1 → 6.1.0](https://github.com/PyCQA/isort/compare/6.0.1...6.1.0)
- https://github.com/psf/black → https://github.com/psf/black-pre-commit-mirror

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

LOCEANlloydizard committed:
fix(ci): free disk space to resolve "No space left on device" during Docker build (#1556)

Added cleanup steps (free-disk-space and docker prune) in docker GitHub Actions workflow
to prevent System.IO.IOException: "No space left on device" errors during multi-arch builds.

dbashford-NOAA committed:
[1542] Support python 3.13 (#1553)

* [1542] Support python 3.13

* [1542] More to support python 3.13

* [1542] Remove netCDF4 max version restriction

LOCEANlloydizard committed:
Merge remote-tracking branch 'upstream/main'

LOCEANlloydizard committed:
Apply suggestions from code review

Co-authored-by: Wu-Jung Lee <leewujung@gmail.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate (#1559)

updates:
- [github.com/PyCQA/isort: 6.1.0 → 7.0.0](https://github.com/PyCQA/isort/compare/6.1.0...7.0.0)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump actions/setup-python from 5.5.0 to 6.0.0 (#1560)

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 5.5.0 to 6.0.0.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v5.5.0...v6.0.0)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-version: 6.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

LOCEANlloydizard committed:
Updated after review

Adapted docstring of bottom_blackwell and shoal_echoview too

dbashford-NOAA committed:
[1428/1512] Migrate to numpy v2 and zarr v3 with updated test_data path format [all tests ci] (#1531)

* [1428/1512] Migrate to numpy v2 and zarr v3 with updated test_data path formatting

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* [1428/1512] Fix issues with migration to numpy v2 and zarr v3

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* [1428/1512] Fix zarr store delete and create and resolve some migration warnings

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* [1428/1512] Reinstall netcdf in github PR workflow to resolve numpy.dtype size changed error

* [1428-1512] Remove FSStore type from zarr store delete

* [1541] Remove python 3.10 support

* force zarr>=3

* [1531] Minor changes to support zarr v3

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* [1531] Adjust to follow line length limit

* [1531] Test removing github workflow netcdf workaround

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Wu-Jung Lee <leewujung@gmail.com>

LOCEANlloydizard committed:
Review Blackwell docstring: fix indentation and replace 'clamp' with clearer wording

leewujung committed:
small tweak re adaptive Sv threshold

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

LOCEANlloydizard committed:
Update docstrings for transient noise and shoal detection functions (#1554)

* Update docstrings for transient noise and shoal detection functions

Update docstrings for transient noise and shoal detection functions, and add credit to Echopy for the original implementations.

* Apply suggestions from code review

Co-authored-by: Wu-Jung Lee <leewujung@gmail.com>

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Updated after review

Adapted docstring of bottom_blackwell and shoal_echoview too

* Review Blackwell docstring: fix indentation and replace 'clamp' with clearer wording

* small tweak re adaptive Sv threshold

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: Wu-Jung Lee <leewujung@gmail.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

LOCEANlloydizard committed:
CI: Pooch assets + Windows fixes, HTTP/S3 test setup, Windows runner support

LOCEANlloydizard committed:
Restore small comments from original pr.yaml

LOCEANlloydizard committed:
CI: switch to Pooch-managed GitHub assets and remove local test_data (#1562)

* CI: use Pooch-cached GitHub assets and lightweight MinIO/HTTP test services

* modified branch for CI on fork

* CI: check changed files on PR only

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* update pr.yaml to match original

* Clean up imports and pass pre-commit checks

* restore small comments from original pr.yaml

* changes after review

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

leewujung committed:
add push to main to tests workflow in additional to PR

leewujung committed:
update build.yaml to use pooch

leewujung committed:
test data endpoint

leewujung committed:
remove trailing white space

leewujung committed:
change name back to test PR

LOCEANlloydizard committed:
Merge pull request #1566 from leewujung/push_workflow_patch

Use pooch in build workflow

LOCEANlloydizard committed:
Merge branch 'main' of https://github.com/OSOceanAcoustics/echopype

LOCEANlloydizard committed:
ci: fix Pooch asset cache key mismatch in build workflow

LOCEANlloydizard committed:
Update pr.yaml

LOCEANlloydizard committed:
Merge from latest upstream

LOCEANlloydizard committed:
Update conftest.py

LOCEANlloydizard committed:
Merge pull request #1567 from LOCEANlloydizard/ci/fix-asset-cache-key

Ci/fix asset cache key

LOCEANlloydizard committed:
Merge pull request #1563 from LOCEANlloydizard/CI_windows

Integration of Windows CI

dependabot[bot] committed:
chore(deps): bump actions/download-artifact from 5 to 6 (#1568)

Bumps [actions/download-artifact](https://github.com/actions/download-artifact) from 5 to 6.
- [Release notes](https://github.com/actions/download-artifact/releases)
- [Commits](https://github.com/actions/download-artifact/compare/v5...v6)

---
updated-dependencies:
- dependency-name: actions/download-artifact
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump actions/upload-artifact from 4 to 5 (#1569)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 4 to 5.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

LOCEANlloydizard committed:
Update test_ecs_integration.py (#1575)

* Update test_ecs_integration.py

Recent xarray versions (2025.11.0) now preserve attributes by default. Because of this, .identical() fails even when values and coordinates are identical, which broke several ECS tests. We now use .equals(), which ignores attributes and keeps the tests stable.

* Update requirements.txt

Update the code so that all tests and RTD builds pass, following the changes from @ctuguinay
Ref: Unpin jupyter-book in docs/requirements.txt #1570

leewujung committed:
Use official http image and using Pooch to pull data to http server (#1573)

* change pr.yaml and build.yaml to use official http and minio images directly

* revert minio changes

* add load_http_server into setup-services

* make sure to copy to the right location

* pull data to /htdocs/data/ directly

* improve logging

* minor tweak

* remove unused scripts

* minor tweak

* remove use official http in docker-compose.aml, remove http build from docker.yaml

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump actions/checkout from 5 to 6 (#1576)

Bumps [actions/checkout](https://github.com/actions/checkout) from 5 to 6.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v5...v6)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

leewujung committed:
Setup python_requires >=3.11 in setup.cfg (#1561)

dependabot[bot] committed:
chore(deps): bump actions/setup-python from 6.0.0 to 6.1.0

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 6.0.0 to 6.1.0.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v6...v6.1.0)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-version: 6.1.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

LOCEANlloydizard committed:
Merge pull request #1580 from OSOceanAcoustics/dependabot/github_actions/actions/setup-python-6.1.0

chore(deps): bump actions/setup-python from 6.0.0 to 6.1.0

leewujung committed:
update azfp6 sha256 and version tag

LOCEANlloydizard committed:
Merge pull request #1582 from leewujung/update_azfp6_asset

Update azfp6 test data

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate (#1583)

updates:
- [github.com/psf/black-pre-commit-mirror: 25.9.0 → 25.12.0](https://github.com/psf/black-pre-commit-mirror/compare/25.9.0...25.12.0)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump actions/download-artifact from 6 to 7

Bumps [actions/download-artifact](https://github.com/actions/download-artifact) from 6 to 7.
- [Release notes](https://github.com/actions/download-artifact/releases)
- [Commits](https://github.com/actions/download-artifact/compare/v6...v7)

---
updated-dependencies:
- dependency-name: actions/download-artifact
  dependency-version: '7'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump actions/upload-artifact from 5 to 6

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 5 to 6.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v5...v6)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump actions/cache from 4 to 5

Bumps [actions/cache](https://github.com/actions/cache) from 4 to 5.
- [Release notes](https://github.com/actions/cache/releases)
- [Changelog](https://github.com/actions/cache/blob/main/RELEASES.md)
- [Commits](https://github.com/actions/cache/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

LOCEANlloydizard committed:
Merge pull request #1584 from OSOceanAcoustics/dependabot/github_actions/actions/download-artifact-7

chore(deps): bump actions/download-artifact from 6 to 7

LOCEANlloydizard committed:
Merge pull request #1585 from OSOceanAcoustics/dependabot/github_actions/actions/upload-artifact-6

chore(deps): bump actions/upload-artifact from 5 to 6

LOCEANlloydizard committed:
Merge pull request #1586 from OSOceanAcoustics/dependabot/github_actions/actions/cache-5

chore(deps): bump actions/cache from 4 to 5

LOCEANlloydizard committed:
Update test_commongrid_api.py

Closes #1590

LOCEANlloydizard committed:
Update conftest.py

LOCEANlloydizard committed:
Merge branch 'main' into fix_numpy_update_2.4

LOCEANlloydizard committed:
Update conftest.py

LOCEANlloydizard committed:
Merge branch 'fix_numpy_update_2.4' of https://github.com/LOCEANlloydizard/echopype into fix_numpy_update_2.4

LOCEANlloydizard committed:
Update pr.yaml

LOCEANlloydizard committed:
Update pr.yaml

LOCEANlloydizard committed:
Update conftest.py

test to see data structure

LOCEANlloydizard committed:
change data version

changing data version

leewujung committed:
trigger test run after restoring v0.11.1a1 azfp.zip

leewujung committed:
Update AZFP test data in v0.11.1a2 assets (#1593)

* update to use azfp in v0.11.1a2

* substitue Ana06/get-changed-files with tj-actions/changed-files

* add debugging step

* list ek60 content for comparison

* update asset without hidden files and sha256 sum

* remove listing azfp and ek60 content fetched by pooch

* clean up updating to tj-actions

leewujung committed:
Merge branch 'main-upstream' into fix_numpy_update_2.4

LOCEANlloydizard committed:
Merge pull request #1591 from LOCEANlloydizard/fix_numpy_update_2.4

Update test_commongrid_api.py to match numpy update 2.4 and add prints to confest.py to display more info

dependabot[bot] committed:
chore(deps): bump tj-actions/changed-files from 45 to 47

Bumps [tj-actions/changed-files](https://github.com/tj-actions/changed-files) from 45 to 47.
- [Release notes](https://github.com/tj-actions/changed-files/releases)
- [Changelog](https://github.com/tj-actions/changed-files/blob/main/HISTORY.md)
- [Commits](https://github.com/tj-actions/changed-files/compare/v45...v47)

---
updated-dependencies:
- dependency-name: tj-actions/changed-files
  dependency-version: '47'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

LOCEANlloydizard committed:
Merge pull request #1594 from OSOceanAcoustics/dependabot/github_actions/tj-actions/changed-files-47

chore(deps): bump tj-actions/changed-files from 45 to 47

LOCEANlloydizard committed:
Investigating the disk space failure in CI (#1592)

* Update pr.yaml

* Update build.yaml

add to build.yaml

* Update build.yaml

add disk cleanup to build workflow

leewujung commented on echopype/consolidate/api.py:
```diff
@@ -205,11 +205,36 @@ def add_depth(
+                if dim_0 in list(echodata[f"Sonar/Beam_group{idx + 1}"].sizes):
+                    if echodata[f"Sonar/Beam_group{idx + 1}"][dim_0].equals(ds[dim_0]):
+                        beam_group_name = f"Beam_group{idx + 1}"
+                        break
```
The block would fail if the number of channels/transducers in the 2 beamgroups are the same.

I think a cleaner approach is to simply check if `dim_0` is `"channel"` or `frequency_nominal`, since this is _within_ echopype and the first dimension should be one of these 2, or the function should error out. Later in line 225-226 you explicitly do an if-else for these 2 allowed dimension names anyway, so no need to broadening this up to other possible dimension names.

leewujung commented on echopype/consolidate/api.py:
```diff
@@ -456,12 +480,22 @@ def add_splitbeam_angle(
+        if dim_0 not in list(echodata[ed_beam_group].sizes) and "channel" in list(
+            echodata[ed_beam_group].sizes
+        ):
+            echodata[ed_beam_group] = swap_dims_channel_frequency(echodata[ed_beam_group])
```
We don't want to swap "frequency_nominal" with "channel" -- we want to _preserve_ whatever the dimension the input ds has and just add the depth variable to the dataset.

leewujung commented on echopype/tests/consolidate/test_consolidate_integration.py:
```diff
@@ -286,6 +286,63 @@ def test_add_splitbeam_angle(sonar_model, test_path_key, raw_file_name, test_pat
+                                               encode_mode=encode_mode,
+                                               to_disk=False)
+
+    # Check that channel dim has been swapped to frequency_nominal
```
make sure to change this too since we want to _preserve_ whatever the input ds has

leewujung reviewed (changes requested):
@dbashford-NOAA : Sorry it again took a while for me to get to this!

I think there may have been some misunderstandings on what #1488 aims to achieve, but hopefully my comments below and inline would clear it up.

The goal of #1488 is to enable users to use _all_ functions in `ep.consolidate` on Sv dataset regardless of if they have called `swap_dims_channel_frequency` or not.

With this in mind, I have a couple suggestions for this PR:
- for all `ep.consolidate` functions, check if either "channel" or "frequency_nominal" exists in the input `ds`, and error out if not. 
  - I'd suggest making a small utility function for this and returns `dim_0`, so that the same routine can be used across all functions. This would help with future maintenance since everything is in one place.
- make sure all `ep.consolidate` functions have a test that checks if the function would run through on Sv dataset _both before and after_ calling `swap_dims_channel_frequency` -- this embodies the goal of #1488

Also, please pull in all upstream changes to make sure your tests will run with the right assets. We've changed the CI setup pretty significantly in the past month or so. Those changes shouldn't impact the components you're working on, but just in case since I saw some directory changes in this PR.

dbashford-NOAA committed:
Merge remote-tracking branch 'origin/1488_support_frequency_nominal_in_consolidate_functions' into 1488_support_frequency_nominal_in_consolidate_functions

# Conflicts:
#	echopype/tests/consolidate/test_add_depth.py

dbashford-NOAA commented:
@leewujung 
Do you know if there is any test data I could use for `add_depth` and `add_splitbeam_angle` that has the `frequency_nominal` dim instead of `channel`?
I think part of the issue is that I was trying to adapt the existing test data to test with a `frequency_nominal` dim.

leewujung commented:
@dbashford-NOAA : ah I see. Echopype generated echodata object and its derivative like the Sv dataset would always have channel as a dimension initially. It’s only when users use swap_dims_channel_frequency will they get that dimension changed to frequency_nominal. We made that swap function because most people refer to different channels by their central frequency - like “the 120 kHz channel.” 

What we can’t control is when users would do the swap, and therefore we want to get all consolidate functions to be able to work with both the before and after swap data. 

So, you can get that by using the swap function.

dbashford-NOAA committed:
[1488] Remove dim_0 switching inside functions and refine testing for swapped dim_0

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

dbashford-NOAA commented:
@leewujung 
I think I've addressed everything you mentioned. I left the dim swap in the test code just as a way to adapt the data to test when dim_0 is frequency_nominal. Let me know if there's anything else that needs to change.
